### ISSUE TITLE
react/prop-types false positive when using props to access array element

### ISSUE THREAD

**Issue by Justkant** [yannickcr/eslint-plugin-react#1259]
Hi everything is here:
<img width="692" alt="capture d ecran 2017-06-14 a 12 26 59" src="https://user-images.githubusercontent.com/3009753/27130225-ba197c02-5105-11e7-9e5c-2b5cdfd727e8.png">

**Comment by ljharb** [comment]
I don't see the propTypes for `Setup`?

also, please provide code in text form, instead of screenshots :-)

**Comment by Justkant** [comment]
Hi,

1. I understand that you prefere text instead of screenshots. It was just for saving time at work.

2. You have no need to see the propTypes for `Setup` because this is not the problem.

3. As you can see in the screenshot Left & Right are both destructured from steps not from props.

Code:

```javascript
const steps = [
  {
    Left: Component1,
    Right: Component2,
  },
];

const Setup = props => {
  const { Left, Right } = steps[props.current];
  return (
    <div className={style.container}>
      <Left {...props} />
      <Right {...props} />
    </div>
  );
}

Setup.propTypes = {
  current: PropTypes.number.isRequired,
};
```

**Comment by Justkant** [comment]
Tomorrow, I'll try this:

```javascript
const Setup = props => {
  const current = props.current;
  const { Left, Right } = steps[current];
  return (
    <div className={style.container}>
      <Left {...props} />
      <Right {...props} />
    </div>
  );
};
```

To confirm that the problem occurs when using some props as an index key and destructuring.

**Comment by ljharb** [comment]
I definitely think that `const { current } = props;` will fix it, but thanks, I think this makes the bug clear :-)
